### PR TITLE
LPS-190599

### PR DESCRIPTION
**Pull request by icgsbr**
# Jira Link

- [LPS-190599](https://liferay.atlassian.net/browse/LPS-190599)

### CODE REVIEW

**Comment by icgsbr** [comment]
Hi @marcelabc ! For what I've seen the errors accused by ci are not related to this implementation. Can you confirm that?

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/util/DDMFormValuesToFieldsConverterImpl.java [hunk 1]
```diff
@@ -52,6 +52,14 @@ public Fields convert(
 				ddmForm.getDDMFormFields(),
 				ddmFormValues.getDDMFormFieldValuesMap(true)));
 
+		if (ddmStructure.getClassName(
+			).contains(
+				"document.library"
+			)) {
```
Comment (marcelabc) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/util/DDMFormValuesToFieldsConverterImpl.java:58]: Hey @icgsbr 
We must avoid method chaining: method().method();

Comment (marcelabc) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/util/DDMFormValuesToFieldsConverterImpl.java:58]: I think we could improve this condition with this code:
```
if (ddmStructureKey.equals(
		DLVideoConstants.DDM_STRUCTURE_KEY_DL_VIDEO_EXTERNAL_SHORTCUT)) {

			ddmFormValues.setDefaultLocale(ddmForm.getDefaultLocale());
}
```

Comment (marcelabc) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/util/DDMFormValuesToFieldsConverterImpl.java:58]: I notice that you are adding a "very specific" condition in this method. I started to wonder:
-  Did we discover why the problem is happening?
- With this solution, are we really fixing the cause of the issue?

Comment (marcelabc) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/util/DDMFormValuesToFieldsConverterImpl.java:58]: If you need help, let me know!

**Review by marcelabc — CHANGES_REQUESTED** [review]
Hi @icgsbr 
Sorry for the delay in reviewing your PR.
I finished my review. Let me know if you have questions :smile:

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/util/DDMFormValuesToFieldsConverterImpl.java (hunk 1 above)
Comment (icgsbr) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/util/DDMFormValuesToFieldsConverterImpl.java:58]: Thanks @marcelabc ! I'll take a look on it!

**Comment by icgsbr** [comment]
Hi @marcelabc ! I've implemented in another location to make sure it affects only the ddm and not the journal article. I still couldn't see any way to refine the solution, I would need some help with that matter.

**Comment by icgsbr** [comment]
Since this issue is related to how ddm understands and processes translation it can be very tricky. I should remember that the whole forms, ddm and related components will probably be refactored, in terms of language/translation, according to this [comment](https://liferay.atlassian.net/browse/LPP-50284?focusedCommentId=2440025).

**Comment by marcelabc** [comment]
ci:test:sf

**Comment by marcelabc** [comment]
ci:test:relevant

**Comment by icgsbr** [comment]
Hi @marcelabc !

Is this PR ready for being fowarded to Bchan?

**Comment by marcelabc** [comment]
Just started reviewing :)

**Review by marcelabc — APPROVED** [review]
LGTM 🎉

**Comment by marcelabc** [comment]
Sorry for the delay in reviewing it @icgsbr !

**Comment by marcelabc** [comment]
ci:forward

**Pull request closed without merging**